dataprofiler/data_readers/data_utils.py

```diff
@@ -610,3 +612,45 @@ def is_stream_buffer(filepath_or_buffer):
     if isinstance(filepath_or_buffer, (StringIO, BytesIO)):
         return True
     return False
+
+def is_valid_url(url_as_string):
+    """
+    Determines whether a given string is a valid URL
+
+    :param url_as_string: string to be tested if URL 
+    :type url_as_string: str
+    :return: true if string is a valid URL
+    :rtype: boolean
+    """
+    result = urlparse(url_as_string)
```

Comment:
this may not be a string, we don't truly know what is being passed in. Hence we should have a try except to catch when this fails.

Reply:
e.g. a user could pass in BytesIO into Data and this may error, no?